Holovibes/sources/api/transform_api.cc

```diff
@@ -0,0 +1,270 @@
+
+void set_time_transformation(const TimeTransformation value)
+{
+    if (api::get_compute_mode() == Computation::Raw || api::get_time_transformation() == value)
+        return;
+
+    UPDATE_SETTING(TimeTransformation, value);
+    set_z_fft_shift(value == TimeTransformation::STFT);
+    get_compute_pipe()->request(ICS::UpdateTimeTransformationAlgorithm);
+}
+
+void update_time_transformation_size(uint time_transformation_size)
+{
+    if (get_compute_mode() == Computation::Raw || get_import_type() == ImportType::None)
+        return;
+
+    if (time_transformation_size == api::get_time_transformation_size())
+        return;
+
+    if (time_transformation_size < 1)
+        time_transformation_size = 1;
+
+    set_time_transformation_size(time_transformation_size);
+    get_compute_pipe()->request(ICS::UpdateTimeTransformationSize);
+}
+
+void set_x_accu_level(uint x_value)
+{
+    SET_SETTING(X, width, x_value);
+    pipe_refresh();
+}
+
+void set_x_cuts(uint value)
+{
+    if (value < get_fd().width)
+    {
+        SET_SETTING(X, start, value);
+        pipe_refresh();
+    }
+}
+
+void set_y_accu_level(uint y_value)
+{
+    SET_SETTING(Y, width, y_value);
+    pipe_refresh();
+}
+
+void set_y_cuts(uint value)
+{
+    if (value < get_fd().height)
+    {
+        SET_SETTING(Y, start, value);
+        pipe_refresh();
+    }
+}
+
+void set_x_y(uint x, uint y)
+{
+    if (get_compute_mode() == Computation::Raw || get_import_type() == ImportType::None)
+        return;
+
+    if (x < get_fd().width)
+        SET_SETTING(X, start, x);
+
+    if (y < get_fd().width)
+        SET_SETTING(Y, start, y);
+
+    pipe_refresh();
+}
+
+void set_q_index(uint value)
+{
+    SET_SETTING(Q, start, value);
+    pipe_refresh();
+}
+
+void set_q_accu_level(uint value)
+{
+    SET_SETTING(Q, width, value);
+    pipe_refresh();
+}
+
+void set_p_index(uint value)
+{
+    if (get_compute_mode() == Computation::Raw)
+        return;
+
+    if (value >= get_time_transformation_size() || value == 0)
+    {
+        LOG_ERROR("p param has to be between 1 and #img");
+        return;
+    }
+
+    SET_SETTING(P, start, value);
+    pipe_refresh();
+}
+
+void set_p_accu_level(uint p_value)
+{
+    SET_SETTING(P, width, p_value);
+    pipe_refresh();
+}
+
+void check_p_limits()
+{
+    int upper_bound = get_time_transformation_size() - 1;
+
+    if (get_p_accu_level() > upper_bound)
+        api::set_p_accu_level(upper_bound);
+
+    upper_bound -= get_p_accu_level();
+
+    if (upper_bound >= 0 && get_p_index() > static_cast<uint>(upper_bound))
+        api::set_p_index(upper_bound);
+}
+
+void check_q_limits()
+{
+    int upper_bound = get_time_transformation_size() - 1;
+
+    if (std::cmp_greater(get_q_accu_level(), upper_bound))
+        api::set_q_accu_level(upper_bound);
+
+    upper_bound -= get_q_accu_level();
+
+    if (upper_bound >= 0 && get_q_index() > static_cast<uint>(upper_bound))
+        api::set_q_index(upper_bound);
+}
+
+#pragma endregion
+
```

Comment:
Could you add :
 if (api::get_compute_mode() == Computation::Raw)
        return;

Reply:
Bounds check and ApiError return code will be done in another PR. This one is large enough